### PR TITLE
[PKX-014] [Enhancement] Use AppRouter for routes

### PR DESCRIPTION
**Pull request by giantiyan**

### CODE REVIEW

File: lib/features/details_page/details_page.dart [hunk 1]
```diff
@@ -37,16 +37,11 @@ class DetailsPage extends StatelessWidget {
                       child: Column(
                         crossAxisAlignment: CrossAxisAlignment.start,
                         children: [
-                          Text(
-                            (specificPokemon?.name).toString().replaceDash.toTitleCase,
-                            style: textTheme.headline1
-                          ),
+                          Text((specificPokemon?.name).toString().replaceDash.toTitleCase, style: textTheme.headline1),
```
Comment (SaturnoMark) [inline lib/features/details_page/details_page.dart]: trailing

File: lib/features/details_page/details_page_connector.dart [hunk 2]
```diff
@@ -6,18 +6,26 @@ import 'package:pokedex/features/details_page/details_page_vm.dart';
...
 
   @override
   Widget build(BuildContext context) {
     return StoreConnector<AppState, DetailsPageVm>(
       vm: () => DetailsPageVmFactory(),
-      onInit: (store) async => store.dispatch(GetPokemonTypesAction(specificPokemon?.url)),
+      onInit: (store) async => store.dispatch(GetPokemonTypesAction(args.specificPokemon?.url)),
       builder: (context, vm) => vm.pageState.when(
-        (types) => DetailsPage(specificPokemon: specificPokemon, types: types),
+        (types) => DetailsPage(specificPokemon: args.specificPokemon, types: types),
```
Comment (SaturnoMark) [inline lib/features/details_page/details_page_connector.dart]: trailing

File: lib/features/details_page/details_page.dart [hunk 3]
```diff
@@ -37,16 +37,11 @@ class DetailsPage extends StatelessWidget {
...
                           Align(
                             alignment: Alignment.centerRight,
-                            child: Text(
-                              specificPokemon?.url?.pokemonId.toString().formatID ?? '',
-                              style: textTheme.headline1?.copyWith(fontSize: 16)
-                            ),
+                            child: Text(specificPokemon?.url?.pokemonId.toString().formatID ?? '',
+                                style: textTheme.headline1?.copyWith(fontSize: 16)),
```
Comment (ryle-roasa) [inline lib/features/details_page/details_page.dart]: trailing

File: lib/pokedex_app.dart [hunk 4]
```diff
@@ -15,7 +16,11 @@ class PokedexApp extends StatelessWidget {
       store: store,
       child: MaterialApp(
         theme: theme,
-        home: const HomePageConnector(),
+        home: Navigator(
+          key: mainNavigatorKey, // This is the key in the AppRouter
+          initialRoute: HomePageConnector.route, // Initial route
+          onGenerateRoute: AppRouter.generateRoute, // This is the method under AppRouter that generates routes
```
Comment (ryle-roasa) [inline lib/pokedex_app.dart]: remove comments hahaha

File: lib/utilities/app_router.dart [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import 'package:flutter/material.dart';
+import 'package:pokedex/features/details_page/details_page_connector.dart';
+import 'package:pokedex/features/home_page/home_page_connector.dart';
+
+final mainNavigatorKey = GlobalKey<NavigatorState>();
+
+class AppRouter {
+  static Route<dynamic> generateRoute(RouteSettings settings) {
+    switch (settings.name) {
```
Comment (ryle-roasa) [inline lib/utilities/app_router.dart:9]: remove comments

File: lib/utilities/app_router.dart [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      // DetailsPageConnector widget, when the user taps on a Pokemon Tile
+      case DetailsPageConnector.route:
+        return MaterialPageRoute(
+            builder: (_) => DetailsPageConnector(args: settings.arguments as DetailsPageConnectorArgs));
+
+      // Default route
+      default:
+        return MaterialPageRoute(
+          builder: (_) => Scaffold(
+            body: Center(child: Text('No route defined for ${settings.name}')),
```
Comment (ryle-roasa) [inline lib/utilities/app_router.dart:20]: remove trailing

Comment (ryle-roasa) [reply lib/utilities/app_router.dart:20]: nvm

**Review by SaturnoMark — APPROVED** [review]

**Pull request merged**